### ISSUE TITLE
High load test

### ISSUE THREAD

**Issue by maxpmaxp** [maxpmaxp/decaptcher#1]
Need to know how many concurrent requests we can serve with different unicorn worker types and use the best one.

**Comment by sorrat** [comment]
First of all, I note that the errors in the first test of the service, in my opinion, were due to the workers class `sync`. I forgot to change it to `gevent`.

For service's testing I developed [test client](https://gist.github.com/sorrat/6672a3e6ebab088a3872) and changed URLs of upstream-services to stub on localhost:

``` python
import time
from bottle import Bottle

app = Bottle()

@app.route('/', method='POST')
def solve():
    time.sleep(5)
    return '0|0|0|0|0|captcha-code'
```

 Also changed the `worker_class` to `gevent`.

When I tested with a large number of concurrent requests (> 1000) there were errors "the client can't connect to the service because of 'too many open files'". And sometimes the service itself couldn't connect to redis-server for the same reason - 'too many open files'.

For that reason, I edited the code to use the connection pool, rather than creating a new for each client. And changed the settings of the OS limit of open file-descriptors:

```
# /etc/sysctl.conf

fs.file-max = 50000
```

```
# /etc/security/limits.conf

<UID> soft nofile 50000
<UID> hard nofile 50000
```

After this errors was gone. Here is the result of testing
(the same results the test showed for `worker_class = 'eventlet'`):

```
$ python test_client.py
All 10000, concurrency 1000, result 10000
All 10000, concurrency 1500, result 10000
All 10000, concurrency 2000, result 10000
All 10000, concurrency 2500, result 10000
All 10000, concurrency 3000, result 10000
All 10000, concurrency 3500, result 10000
All 10000, concurrency 4000, result 10000
All 10000, concurrency 4500, result 10000
All 10000, concurrency 5000, result 10000
```

Where columns: `<total> <concurrency> <successful>`

For example test for `worker_class = 'sync'` (without using threads)

```
$ python test_client.py
All 1000, concurrency 100, result 1000
All 1000, concurrency 350, result 586
All 1000, concurrency 600, result 492
```

Errors for client's log:

```
Max retries exceeded with url: / (Caused by <class 'socket.error'>:
[Errno 110] Connection timed out)
```

Bottom line: the workers of class `gevent` with proper setting of OS parameters copes with a large number of concurrent requests.
